src/k8s/request/cloudInit.js

```diff
@@ -0,0 +1,69 @@
+import { safeDump } from 'js-yaml';
+import { CLOUDINIT_DISK } from '../../constants';
+
+const resolveScriptHeader = userData => {
+  const validScriptHeaders = [
+    'Content-Type',
+    '#include',
+    '#!',
+    '#cloud-config',
+    '#upstart-job',
+    '#part-handler',
+    '#cloud-boothook',
+  ];
+
+  if (
+    typeof userData !== 'string' ||
+    userData.length === 0 ||
+    validScriptHeaders.find(header => userData.startsWith(header))
+  ) {
+    return userData;
+  }
+
+  return `#cloud-config\n${userData}`;
+};
+
+export class CloudInit {
+  constructor(props) {
+    this.disk = (props && props.disk) || {
+      name: CLOUDINIT_DISK,
+    };
+
+    this.volume = (props && props.volume) || {
+      name: CLOUDINIT_DISK,
+      cloudInitNoCloud: {},
+    };
+  }
+
+  setUserData(userData) {
+    this.volume.cloudInitNoCloud.userData = userData;
+  }
+
+  setPredefinedUserData({ hostname, sshAuthorizedKeys }) {
+    const userDataObject = {};
+    if (sshAuthorizedKeys) {
+      userDataObject.users = [
+        {
+          name: 'default',
```

Comment:
note: I chose to use default account intstead of root which might not be enabled by default

Reply:
lgtm
Can you please add similar comment to the code?